### PR TITLE
Debug task

### PR DESCRIPTION
**Pull request by dgreene-r7**
This PR adds a `debug-server` npm task that spins up test metadata and test S3 servers and runs an instance of propsd using a dev config file.

### CODE REVIEW

**Comment by rvanburen-r7** [comment]
LGTM, but you should get another set of :eyes: on it besides mine.

File: lib/config.js [hunk 1]
```diff
@@ -28,10 +27,12 @@ const defaults = {
...
 function loadConfigs(path) {
-  const configPath = path || Path.resolve(__dirname, '../config/dev.json');
-  const configs = nconf.file(configPath);
+  let configs = nconf.defaults(defaults);
+
+  if (path) {
+    configs = nconf.file(path);
+  }
 
-  configs.defaults(defaults);
```
Comment (fmitchell-r7) [inline lib/config.js]: Looks like we lost the behavior where defaults got filled in for config bits that weren't in the file. Was that intentional?

File: test/bin/s3-server.js [hunk 2]
```diff
@@ -73,55 +83,101 @@ function createTempDirForBucket() {
...
 
+/**
+ * Handles cleanup on app exit
+ */
 function exitHandler() {
-  console.log('Cleaning up temp directory'); // eslint-disable-line no-console
+  Log.info('Cleaning up temp directory');
   cleanupTempDir();
   process.exit(0);
 }
```
Comment (fmitchell-r7) [inline test/bin/s3-server.js]: Do we need API docs for the `createBucket` function so it matches up with the `emptyBucket` function?

File: lib/config.js (hunk 1 above)
Comment (dgreene-r7) [inline lib/config.js]: See line 30

Comment (dgreene-r7) [inline lib/config.js]: Good catch. I moved it to line 30 but forgot to change variable names.

**Comment by fmitchell-r7** [comment]
Thanks for adding this! Is it worth putting a note in the README [like we have for the S3 server](https://github.com/rapid7/propsd#s3-server) so people know this is available for local integration testing?

File: lib/config.js [hunk 3]
```diff
@@ -28,10 +27,12 @@ const defaults = {
 };
 
 function loadConfigs(path) {
-  const configPath = path || Path.resolve(__dirname, '../config/dev.json');
-  const configs = nconf.file(configPath);
+  let configs = nconf.defaults(defaults);
```
Comment (jmanero-r7) [inline lib/config.js]: This is a weird usage of nconf. Generally, configuration is a singleton, so the nconf package actually exports an instantiated config object, which you attach sources to:

```javascript

var Config = globals.Confog = require('nconf');
Config.argv(...);
Config.file(...);
Config.defaults({
...
});

module.exports = Config;
```

**Comment by dgreene-r7** [comment]
I'm going to close this PR, fix the config service, and then reopen.

**Pull request closed without merging**